mtags/src/main/scala/scala/meta/internal/semver/SemVer.scala

```diff
@@ -2,18 +2,26 @@ package scala.meta.internal.semver
 
 object SemVer {
 
-  def isCompatibleVersion(minimumVersion: String, version: String): Boolean = {
+  case class Version(major: Int, minor: Int, patch: Int) {
+    def >(that: Version): Boolean = {
+      this.major > that.major ||
+      (this.major == that.major && this.minor > that.minor) ||
+      (this.major == that.major && this.minor == that.minor && this.patch > that.patch)
+    }
+
+    def >=(that: Version): Boolean = this > that || this == that
+  }
 
-    def splitVersion(v: String) =
-      v.replaceAll("(-|\\+).+$", "").split('.')
-    val minVersionSplit = splitVersion(minimumVersion).map(_.toInt)
-    val versionSplit = splitVersion(version).map(_.toInt)
-    (minVersionSplit, versionSplit) match {
-      case (Array(minMajor, minMinor, minPatch), Array(major, minor, patch)) =>
-        (major > minMajor) ||
-          (major == minMajor && minor > minMinor) ||
-          (major == minMajor && minor == minMinor && patch >= minPatch)
-      case _ => false
+  object Version {
+    def fromString(version: String): Version = {
+      val Array(major, minor, patch) =
+        version.replaceAll("(-|\\+).+$", "").split('.').map(_.toInt)
+
+      Version(major, minor, patch)
     }
   }
+
+  def isCompatibleVersion(minimumVersion: String, version: String): Boolean = {
+    Version.fromString(version) >= Version.fromString(minimumVersion)
+  }
```

Comment:
I refactored this to support a slightly more general use case.

We still compare versions only by x.y.z, ignoring any other following modifier. This has some edge cases such as `1.0.0+1` being equal to `1.0.0-SNAPSHOT` and `1.0.0+56-abcdefg`, but it still covers all our uses cases for now